ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -13,114 +13,330 @@
   }
 
-  IndexValue(long size, Offset offset) {
-    this(size, offset, (byte) 0, Utils.Infinite_Time);
+  /**
+   * Constructs IndexValue based on the args passed
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param flags the {@link Flags} that needs to be set for the Index Value
+   * @param expiresAtMs the expiration time in ms at which the blob expires
+   * @param operationTimeInMs operation time in ms of the entry
+   * @param serviceId the serviceId that this blob belongs to
+   * @param containerId the containerId that this blob belongs to
+   */
+  IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long operationTimeInMs, short serviceId,
+      short containerId) {
+    this(size, offset, flags, expiresAtMs, offset.getOffset(), operationTimeInMs, serviceId, containerId);
   }
 
-  private IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long originalMessageOffset) {
+  /**
+   * Constructs IndexValue based on the args passed
+   * @param size the size of the blob that this index value refers to
+   * @param offset the {@link Offset} in the {@link Log} where the blob that this index value refers to resides
+   * @param flags the flags that needs to be set for the Index Value
+   * @param expiresAtMs the expiration time in ms at which the blob expires
+   * @param originalMessageOffset the original message offset where the Put record pertaining to a delete record exists
+   *                              in the same log segment. Set to -1 otherwise.
+   * @param operationTimeInMs operation time in ms of the entry in secs
+   * @param serviceId the serviceId that this blob belongs to
+   * @param containerId the containerId that this blob belongs to
+   */
+  private IndexValue(long size, Offset offset, byte flags, long expiresAtMs, long originalMessageOffset,
+      long operationTimeInMs, short serviceId, short containerId) {
+    this.size = size;
     this.offset = offset;
-    value = ByteBuffer.allocate(Index_Value_Size_In_Bytes);
-    value.putLong(size);
-    value.putLong(offset.getOffset());
-    value.put(flags);
-    value.putLong(expiresAtMs);
-    value.putLong(originalMessageOffset);
-    value.position(0);
+    this.flags = flags;
+    this.expiresAtMs = getTimeInMsToTheNearestSec(expiresAtMs);
+    this.originalMessageOffset = originalMessageOffset;
+    this.operationTimeInMs = getTimeInMsToTheNearestSec(operationTimeInMs);
+    this.serviceId = serviceId;
+    this.containerId = containerId;
+    version = PersistentIndex.CURRENT_VERSION;
+  }
+
+  /**
+   * Get time in ms to the nearest second(floored) for the given time in ms
+   * @param timeInMs the time in ms that needs to be converted
+   * @return the time in ms to the nearest second(floored) for the given time in ms
+   */
+  private long getTimeInMsToTheNearestSec(long timeInMs) {
```

Comment:
maybe this can be a function in `Utils`? Looks like `IndexSegment` and lots of tests also use this